fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -50,24 +67,39 @@
 
     /**
      * Basic constructor.
-     * @param fullId The full identifier.
+     * @param fullId The full identifier or null if root.
      * @throws IllegalArgumentException If ID does not start with expected prefix.
      */
-    public FedoraID(final String fullId) {
-        if (!fullId.startsWith(FEDORA_ID_PREFIX)) {
-            throw new IllegalArgumentException(String.format("ID must begin with %s", FEDORA_ID_PREFIX));
-        }
-        this.fullId = fullId;
+    private FedoraID(final String fullId) {
+        this.fullId = ensurePrefix(fullId);
         if (!this.fullId.equals(FEDORA_ID_PREFIX)) {
             // Only strip trailing slashes the ID is more than the info:fedora/ prefix.
             this.fullId = this.fullId.replaceAll("/+$", "");
         }
         // Carry the path of the request for any exceptions.
-        this.externalPath = fullId.substring(FEDORA_ID_PREFIX.length());
+        this.externalPath = this.fullId.substring(FEDORA_ID_PREFIX.length());
```

Comment:
Yes it is the external path of the request which is the same as the full id without the FEDORA_ID_PREFIX (but with a leading / added). If that is confusing I could make a private method that just strips `info:fedora` off the front of the `fullId`

Reply:
Might mainly be the word "external" that's throwing me off since we use it elsewhere to describe the form of an id presented at the HTTP layer, like "http://localhost:8080/fcrepo/rest/12345". Whether it comes from a precalculated variable or a helper method is okay with me.